src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/ApiConventionItem.cs

```diff
@@ -0,0 +1,199 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+
+namespace Microsoft.AspNetCore.Mvc.Infrastructure
+{
+    /// <summary>
+    /// Metadata associated with an action method via API convention.
+    /// </summary>
+    public sealed class ApiConventionItem
+    {
+        public ApiConventionItem(IReadOnlyList<IApiResponseMetadataProvider> responseMetadataProviders)
```

Comment:
@rynowak this is public. It does let users do their own thing (https://github.com/aspnet/Mvc/issues/7906) if they choose to, but maybe that's ok. The alternative is to use a `.Internal` thing instead since the type's shared between .Core and ApiExplorer

Reply:
I'm not in love with the name, but the type definition seems fine. Why does this need to be defined in .Core?

Reply:
We calculate it as part of the `ApiBehaviorApplicationModel`. Part of doing this is so we can flag invalid API conventions at app start.

Reply:
Suggestions for alternative names? `ApiConventionMetadata` \ `ApiConventionResult`?